internal/sidecar/service/reclaimspace.go

```diff
@@ -212,19 +212,14 @@ func (rs *ReclaimSpaceServer) getStagingTargetPath(pv *corev1.PersistentVolume)
 		return "", fmt.Errorf("failed to detect Kubernetes version: %w", err)
 	}
 
-	major, err := strconv.Atoi(version.Major)
+	versionString := fmt.Sprintf("v%s.%s", version.Major, version.Minor)
+	semanticVersion, err := semver.NewVersion(versionString)
 	if err != nil {
-		return "", fmt.Errorf("failed to convert Kubernetes major version %q to int: %w", version.Major, err)
+		return "", fmt.Errorf("failed to convert Kubernetes version %q to semver object: %w", version, err)
 	}
 
-	minor, err := strconv.Atoi(version.Minor)
-	if err != nil {
-		return "", fmt.Errorf("failed to convert Kubernetes minor version %q to int: %w", version.Minor, err)
-	}
-
-	// 'encode' major/minor in a single integer
-	legacyVersion := 1024 // Kubernetes 1.24 => 1 * 1000 + 24
-	if ((major * 1000) + minor) < (legacyVersion) {
+	legacyVersion, _ := semver.NewVersion("v1.24")
+	if semanticVersion.LessThan(legacyVersion) {
```

Comment:
can we use `https://pkg.go.dev/k8s.io/apimachinery/pkg/util/version#Version.LessThan` package for this one?  we don't need to extract the version and do compare by ourself.

Reply:
Good idea, I've tried switching to that version and got rid of the vendored items.

Reply:
@psy-q Sorry again i missed adding one point, we can remove this entire check as we don't need to support 1.24 anyone as its deprecated or unsupported Kubernetes version. IMO we can remove all these checks

Reply:
@psy-q @nixpanic instead of getting it fixed why not remove it as its unsupported version?

Reply:
Ah, that's good news, I will remove the check and the legacy global mount section.

Reply:
I've removed the check. I noticed that the way the function is called, it checks for errors but `err` is always `nil` for `getStagingTargetPath()`. Should I remove the error check as well?